src/main/java/org/mitre/synthea/world/concepts/Claim.java

```diff
@@ -0,0 +1,188 @@
+package org.mitre.synthea.world.concepts;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mitre.synthea.engine.Module;
+import org.mitre.synthea.world.agents.Payer;
+import org.mitre.synthea.world.agents.Person;
+import org.mitre.synthea.world.agents.Provider;
+import org.mitre.synthea.world.concepts.HealthRecord.Encounter;
+import org.mitre.synthea.world.concepts.HealthRecord.Entry;
+import org.mitre.synthea.world.concepts.HealthRecord.Medication;
+
+public class Claim {
+
+  private Encounter encounter;
+  private Medication medication;
+  // The Entries have the actual cost, so the claim has the amount that the payer covered.
+  private double coveredCost;
+  public Payer payer;
+  public Person person;
+  public List<Entry> items;
+
+  /**
+   * Constructor of a health insurance Claim for an encounter.
+   */
+  public Claim(Encounter encounter, Person person) {
+    this((Entry) encounter, person);
+    this.encounter = encounter;
+  }
+
+  /**
+   * Constructor of a health insurance Claim for a medication.
+   */
+  public Claim(Medication medication, Person person) {
+    this((Entry) medication, person);
+    this.medication = medication;
+  }
+
+  /**
+   * Constructor of a Claim for an Entry.
+   */
+  private Claim(Entry entry, Person person) {
+    // Set the Person.
+    this.person = person;
+    // Set the Payer.
+    if (this.person.getPayerAtTime(entry.start) == null) {
+      // Person hasn't checked to get insurance at this age yet. Will check now.
+      Module.processHealthInsuranceModule(this.person, entry.start);
+    }
+    this.payer = this.person.getPayerAtTime(entry.start);
+    this.items = new ArrayList<Entry>();
+  }
+
+  /**
+   * Default constructor to create a blank claim that will be unused because
+   * health insurance is turned off.
+   */
+  public Claim() {
+    this.items = new ArrayList<Entry>();
+  }
+
+  /**
+   * Adds non-explicit costs to the Claim. (Procedures/Immunizations/etc).
+   */
+  public void addLineItem(Entry entry) {
+    this.items.add(entry);
+  }
+
+  /**
+   * Assigns the costs of the claim to the patient and payer.
+   */
+  public void assignCosts() {
+
+    // Casts either the encounter or medication of the claim depending on which is not null.
+    Entry entry;
+    Provider provider = null;
+
+    if (this.encounter != null) {
```

Comment:
Can't lines 79 - 92 just be:

```
double totalCost = getTotalClaimCost();
```

Reply:
I ended up refactoring Claims a bit and removed the encounter and medication fields for it, replacing it with just a field mainEntry. This eliminates the checking and shortens/simplifies Claim code.